src/Sparrow/Json/BlittableJsonReaderObject.cs

```diff
@@ -483,5 +485,180 @@ public void CopyTo(byte* ptr)
+        }
+
+        private BlittableJsonToken TokenValidation(byte tokenStart, out int propOffsetSize,
+            out int propIdSize)
+        {
+            var token = (BlittableJsonToken)tokenStart;
+            var tokenType = ProcessTokenTypeFlags(token);
+            propOffsetSize = ((tokenType == BlittableJsonToken.StartObject) ||
+                               (tokenType == BlittableJsonToken.StartArray))
+                ? ProcessTokenOffsetFlags(token)
+                : 0;
+
+            propIdSize = (tokenType == BlittableJsonToken.StartObject)
+                ? ProcessTokenPropertyFlags(token)
+                : 0;
+            return tokenType;
+        }
+
+        private int PropertiesValidation(BlittableJsonToken rootTokenTypen, int mainPropOffsetSize, int mainPropIdSize,
+            int objStartOffset, int numberOfPropsNames)
+        {
+            byte offset;
+            var numberOfProperties = ReadVariableSizeInt(_mem + objStartOffset, 0, out offset);
+            var current = objStartOffset + 1;
+
+            if (numberOfProperties < 0)
+                throw new InvalidDataException("Number of properties not valid");
+
+            for (var i = 1; i <= numberOfProperties; i++)
+            {
+                var propOffset = ReadNumber(_mem + current, mainPropOffsetSize);
+                if ((propOffset > objStartOffset) || (propOffset < 1))
+                    throw new InvalidDataException("Properties offset not valid");
+                current += mainPropOffsetSize;
+
+                if (rootTokenTypen == BlittableJsonToken.StartObject)
+                {
+                    var id = ReadNumber(_mem + current, mainPropIdSize);
+                    if ((id > numberOfPropsNames) || (id < 0))
+                        throw new InvalidDataException("Properties id not valid");
+                    current += mainPropIdSize;
+                }
+
+                int propOffsetSize;
+                int propIdSize;
+                var tokenType = TokenValidation(*(_mem + current), out propOffsetSize, out propIdSize);
+                current++;
+
+                var propValueOffset = objStartOffset - propOffset;
+                switch (tokenType)
+                {
+                    case BlittableJsonToken.StartObject:
+                        PropertiesValidation(tokenType, propOffsetSize, propIdSize, propValueOffset, numberOfPropsNames);
+                        break;
+                    case BlittableJsonToken.StartArray:
+                        PropertiesValidation(tokenType, propOffsetSize, propIdSize, propValueOffset, numberOfPropsNames);
+                        break;
+                    case BlittableJsonToken.Integer:
+                        ReadVariableSizeIntInReverse(_mem, propValueOffset, out offset);
+                        break;
+                    case BlittableJsonToken.Float:
+                        var floatLen = ReadNumber(_mem + objStartOffset - propOffset, 1);
```

Comment:
This isn't validating that this is an actual float. You need to parse it to actually verify it.